OWML.ModLoader/ModData.cs

```diff
@@ -21,17 +22,42 @@ public class ModData : IModData
                                  Config != null && Config.RequireVR ||
                                  Config == null && DefaultConfig != null && DefaultConfig.RequireVR;
 
+        private IModConfig _configSnapshot;
+
         public ModData(IModManifest manifest, IModConfig config, IModConfig defaultConfig)
         {
             Manifest = manifest;
             Config = config;
             DefaultConfig = defaultConfig;
+            UpdateSnapshot();
+        }
+
+        public void UpdateSnapshot()
+        {
+            if (Config != null)
+            {
+                _configSnapshot = Config.Copy();
+            }
+            else
+            {
+                _configSnapshot = DefaultConfig?.Copy();
+            }
         }
 
         public void ResetConfigToDefaults()
         {
-            Config.Enabled = DefaultConfig.Enabled;
-            Config.Settings = new Dictionary<string, object>(DefaultConfig.Settings);
+            if (DefaultConfig != null)
+            {
+                Config = DefaultConfig.Copy();
+            }
+            else if (Config != null)
+            {
+                Config.Enabled = true; //to preserve other settings
+            }
+            else
+            {
+                Config = new ModConfig();
+            }
```

Comment:
why do you wanna preserve other settings?
i suggest:
`Config = DefaultConfig != null ? DefaultConfig.Copy() : new ModConfig();`